src/m365/aad/commands/sp/sp-list.spec.ts

```diff
@@ -0,0 +1,201 @@
+        description: null,
+        disabledByMicrosoftStatus: null,
+        displayName: "My custom service principal",
+        homepage: null,
+        loginUrl: null,
+        logoutUrl: null,
+        notes: null,
+        notificationEmailAddresses: [],
+        preferredSingleSignOnMode: null,
+        preferredTokenSigningKeyThumbprint: null,
+        replyUrls: [
+          "urn:ietf:wg:oauth:2.0:oob",
+          "https://localhost",
+          "http://localhost",
+          "http://localhost:8400"
+        ],
+        servicePrincipalNames: [
+          "https://contoso.onmicrosoft.com/907a8cea-411a-461a-bb30-261e52febcca",
+          "907a8cea-411a-461a-bb30-261e52febcca"
+        ],
+        servicePrincipalType: "Application",
+        signInAudience: "AzureADMultipleOrgs",
+        tags: [
+          "WindowsAzureActiveDirectoryIntegratedApp"
+        ],
+        tokenEncryptionKeyId: null,
+        samlSingleSignOnSettings: null,
+        addIns: [],
+        appRoles: [],
+        info: {
+          logoUrl: null,
+          marketingUrl: null,
+          privacyStatementUrl: null,
+          supportUrl: null,
+          termsOfServiceUrl: null
+        },
+        keyCredentials: [],
+        oauth2PermissionScopes: [
+          {
+            adminConsentDescription: "Allow the application to access My custom service principal on behalf of the signed -in user.",
+            adminConsentDisplayName: "Access My custom service principal",
+            id: "907a8cea-411a-461a-bb30-261e52febcca",
+            isEnabled: true,
+            type: "User",
+            userConsentDescription: "Allow the application to access My custom service principal on your behalf.",
+            userConsentDisplayName: "Access My custom service principal",
+            value: "user_impersonation"
+          }
+        ],
+        passwordCredentials: [],
+        resourceSpecificApplicationPermissions: [],
+        verifiedPublisher: {
+          displayName: null,
+          verifiedPublisherId: null,
+          addedDateTime: null
+        }
+      }
+    ]
+  };
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(telemetry, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    sinon.stub(session, 'getId').callsFake(() => '');
+    auth.service.connected = true;
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName,
+      session.getId
+    ]);
```

Comment:
Hi, just popping in here to say that we should refactor this `sinonUtil.restore([...` to `sinon.restore()`. This a new way to ensure that the entire sandbox is cleared and will from now on be used in every command.